src/counter.rs

```diff
@@ -0,0 +1,341 @@
+    async fn counters_order_test() {
+        let c = Counters::start(&vec![
+            "ai-runtime.requests".into(),
+            "golem.usage.duration_sec".into(),
+            "golem.usage.gpu-sec".into(),
+        ])
+        .expect("Creates counters");
+        let counters = c.counters.read().await;
+        assert!(matches!(
+            counters.first(),
+            Some(&SupportedCounter::RequestsCount(..))
+        ));
+        assert!(matches!(
+            counters.get(1),
+            Some(&SupportedCounter::Duration(..))
+        ));
+        assert!(matches!(
+            counters.get(2),
+            Some(&SupportedCounter::RequestsDuration(..))
+        ));
+        assert_eq!(counters.len(), 3);
+    }
+
+    #[tokio::test]
+    async fn one_counter_test() {
+        let counters =
+            Counters::start(&vec!["golem.usage.duration_sec".into()]).expect("Creates counters");
+        let counters = counters.counters.read().await;
+        assert!(matches!(
+            counters.first(),
+            Some(&SupportedCounter::Duration(..))
+        ));
+        assert_eq!(counters.len(), 1);
+    }
+
+    #[tokio::test]
+    async fn zero_counter_error_test() {
+        let counters = Counters::start(&vec![]).expect("Creates empty counters collection");
+        let counters = counters.counters.read().await;
+        assert!(counters.is_empty());
+    }
+
+    #[tokio::test]
+    async fn unsupported_counter_test() {
+        let res = Counters::start(&vec![
+            "golem.usage.duration_sec".into(),
+            "unknown.timer.fail".into(),
+        ]);
+        assert!(matches!(res, Err(err) if err.to_string().contains("Unsupported counter")));
+        let res = Counters::start(&vec!["unknown.timer.fail".into()]);
+        assert!(matches!(res, Err(err) if err.to_string().contains("Unsupported counter")));
+    }
+
+    #[tokio::test]
+    async fn overlapping_requests_counter_test() {
+        let counters = Counters::start(&vec![
+            "golem.usage.duration_sec".into(),
+            "ai-runtime.requests".into(),
+            "golem.usage.gpu-sec".into(),
+        ])
+        .expect("Creates counters");
+
+        let test_tasks = task::LocalSet::new();
+
+        let delay = std::time::Duration::from_secs(1);
+        let mut requests_monitor = counters.requests_monitor();
+
+        // 3 requests at step 1, 3, and 5.
+        test_tasks.spawn_local(async move {
+            for i in 0..6 {
+                // println!("Step: {i}");
+                if (i & 1) == 1 {
+                    // odd step number
+                    // println!("Short request on step: {i}");
+                    let response_monitor = requests_monitor.on_request().await;
+                    // println!("Short request on step: {i}. Done.");
+                    tokio::time::sleep(delay).await;
+                    response_monitor.on_response();
+                    // println!("Short request response on step: {i}");
```

Comment:
Please remove unused prints here and in the rest of tests

Reply:
I left them because test verifies time measurement which makes stoping with debugger difficult. Printing to console every single time is unnecessary. Of course I can remove these commented prints.